### DISCUSSION TITLE
Should historical data be provided by the platform for more than 1,000 data points

### DISCUSSION THREAD

**Discussion by microprediction** [microprediction/microprediction#79] [General]
Various solutions are possible. Some discussion points:

- https://github.com/microprediction/history could be fixed pretty easily. Anyone want to take a stab? 
- Other warehousing is possible, but doing on on the same architecture is probably not a good idea. Microprediction.Org is at heart a short-term cache (quite literally, actually) for performance reasons. 
- Many algorithms could be more efficient in data transfer as it is, for example the call to lagged_values could specify an earliest time and the micromanager could staple time-series together as it goes. 
- It isn't possible to anticipate which time-series justify more than 1,000 data points, as compared with global forecasting for example. 
- It is possible for stream creators to establish various patterns motivated by backtesting (longer discussion)
- There's some discussion in the book too.

**Comment by spolgul** [comment]
Here is a possible implementation: https://arrow.apache.org/docs/python/ipc.html

filesystem interface can be local or remote/s3, etc: https://arrow.apache.org/docs/python/filesystems.html
